### PR TITLE
chore(website): auto-inject ban-types default options

### LINKED ISSUE #6210: Docs: ban-types.md should pull in defaultTypes automatically
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

https://github.com/typescript-eslint/typescript-eslint/pull/6209: `ban-types.md` has a copy & pasted description of the default rule options. We should instead pull from the source file automatically.

@orta suggested https://github.com/orta/markdown-magic-inline-types & co. as a potential option. I also would love to see if there's anything in Docusaurus for this kind of feature request.

### Affected URL(s)

https://typescript-eslint.io/rules/ban-types/

### PR DESCRIPTION
**Pull request by Josh-Cena**
## PR Checklist

- [x] Addresses an existing open issue: fixes #6210
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Ideally I should use an actual parser to get the `defaultTypes`, but there's no parser package readily installed in the website, so I thought this should do too.

### CODE REVIEW

File: packages/website/plugins/generated-rule-docs.ts
Comment (JoshuaKGoldberg) [inline packages/website/plugins/generated-rule-docs.ts:303]: 😬 I do dislike having such rule-specific things in this already kind of big file... what do you think about having an mdx component like `<BanTypesDefaultOptions />`?

Comment (Josh-Cena) [reply packages/website/plugins/generated-rule-docs.ts:303]: We already have file-specific handling for this page. This cannot be an MDX component because it has to work server-side and get statically injected instead of dynamically rendered.

Comment (JoshuaKGoldberg) [reply packages/website/plugins/generated-rule-docs.ts:303]: Blurgh, I keep forgetting these things. MDX components don't work server-side? Is this mentioned in docs / one of the things blocked by MDX v2? (I searched and can't find it)

**Review by JoshuaKGoldberg — APPROVED** [review]
Seems reasonable - I'll defer to you on Docusaurus things 😄

File: packages/website/plugins/generated-rule-docs.ts
Comment (Josh-Cena) [reply packages/website/plugins/generated-rule-docs.ts:303]: MDX components are basically React components. They can get server-rendered but they can't access server data like files.

**Pull request merged**
